packages/ember-data/lib/system/relationships/state/belongs-to.js

```diff
@@ -60,14 +60,18 @@ BelongsToRelationship.prototype._super$addRecord = Relationship.prototype.addRec
 BelongsToRelationship.prototype.addRecord = function(newRecord) {
   if (this.members.has(newRecord)) { return;}
   var type = this.relationshipMeta.type;
-  Ember.assert("You cannot add a '" + newRecord.constructor.typeKey + "' record to the '" + this.record.constructor.typeKey + "." + this.key +"'. " + "You can only add a '" + type.typeKey + "' record to this relationship.", (function () {
-    if (type.__isMixin) {
-      return type.__mixin.detect(newRecord);
+  var modelClass = this.store.modelFor(type);
+  Ember.assert("You cannot add a '" + newRecord.constructor.typeKey + "' record to the '" + this.record.constructor.typeKey + "." + this.key +"'. " + "You can only add a '" + modelClass.typeKey + "' record to this relationship.", (function () {
+    if (modelClass.__isMixin) {
+      return modelClass.__mixin.detect(newRecord);
     }
-    if (Ember.MODEL_FACTORY_INJECTIONS) {
-      type = type.superclass;
+
+    // TODO: There seems to be a weird cylic error with loader js here.
+    if (modelClass.superclass && modelClass.superclass !== DS.Model) {
```

Comment:
Need to talk to stef here. This inheritance check is wacky. Not sure what to check for wrt injections.